main.go

```diff
@@ -34,7 +34,6 @@ func main() {
 	var mgr ctrl.Manager
 
 	listeningPort := flag.Uint("listening-port", 9001, "HTTP port the proxy listens to (default: 9001)")
-	k8sControlPlaneUrl := flag.String("k8s-control-plane-url", "https://kubernetes.default.svc", "Kubernetes control plane URL (default: https://kubernetes.default.svc)")
```

Comment:
We have to keep this for backward compatibility, planning to remove starting from 0.1.0.

Let's default to an empty string and add a log informing the end-user we're going to deprecate this in the upcoming release. If the value is provided we'll use it.